cli/src/breakpoint_utils.py

```diff
@@ -0,0 +1,273 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module provides a variety of utilities related to breakpoint processing.
+
+These utilities are useful in multiple snapshot and logpoint commands.
+"""
+
+import datetime
+import re
+import time
+
+from exceptions import SilentlyExitError
+
+# Regex that can be used to validate a user inputted location which should be in
+# the format file:line.
+LOCATION_REGEX = '^[^:]+:[1-9][0-9]*$'
+
+# Impose some 'reasonable' max, here we choose 2^31-1, won't cause any overflows
+# anywhere and is more than enough for a file's line number.
+MAX_LINE_NUMBER = 2147483647
```

Comment:
I hope that nobody has files with that many lines...

In all honesty though, are there limitations elsewhere in the system that are more restrictive than this number?

Reply:
I'm not aware of any other limitations, if there were however I'd expect the request to simply fail. The Pantheon UI does not impose any lower restriction than this. I felt there should be some sanity check on the value, and this seemed like a safe upper bound.